src/front/glsl/variables.rs

```diff
@@ -0,0 +1,93 @@
+use crate::{
+    Binding, BuiltIn, Expression, GlobalVariable, Handle, ScalarKind, ShaderStage, StorageAccess,
+    StorageClass, Type, TypeInner, VectorSize,
+};
+
+use super::ast::*;
+use super::error::ErrorKind;
+
+impl Program {
+    pub fn lookup_variable(&mut self, name: &str) -> Result<Option<Handle<Expression>>, ErrorKind> {
+        let mut expression: Option<Handle<Expression>> = None;
+        match name {
+            "gl_Position" => {
+                #[cfg(feature = "glsl-validate")]
+                match self.shader_stage {
+                    ShaderStage::Vertex | ShaderStage::Fragment { .. } => {}
+                    _ => {
+                        return Err(ErrorKind::VariableNotAvailable(name.into()));
+                    }
+                };
```

Comment:
Maybe there is a better way than `match`, I tried `if ! let`, but that doesn't work.